### PR TITLE
Cleanup get cell by id call

### PR DESCRIPTION
**Pull request by sulochan**

### CODE REVIEW

**Comment by sulochan** [comment]
WIP

**Comment by sulochan** [comment]
Good for review / merge.

File: craton/inventory/tests/unit/test_api.py [hunk 1]
```diff
@@ -40,6 +40,19 @@ def delete(self, path):
...
+    @mock.patch.object(dbapi, 'cells_get_by_id')
+    def test_get_cells_by_id(self, mock_cells):
+        mock_cells.return_value = fake_resources.CELL1
+        resp = self.get('v1/cells/1')
+        self.assertEqual(resp.json["name"], fake_resources.CELL1.name)
+
+    @mock.patch.object(dbapi, 'cells_delete')
+    def test_cells_delete(self, mock_cell):
+        resp = self.delete('v1/cells/1')
+        self.assertEqual(200, resp.status_code)
```
Comment (jimbaker) [inline craton/inventory/tests/unit/test_api.py]: What if we tested double deletes so we can get that edge case tested?

File: craton/inventory/api/v1/cells.py [hunk 2]
```diff
@@ -73,6 +73,23 @@ def post(self):
...
+class CellById(base.Resource):
+
+    def get(self, id):
+        context = request.environ.get('context')
+        try:
+            cell_obj = dbapi.cells_get_by_id(context, id)
+        except exceptions.NotFound:
+            return self.error_response(404, 'Not Found')
+        except Exception as err:
+            LOG.error("Error during Cell get by id: %s" % err)
```
Comment (jimbaker) [inline craton/inventory/api/v1/cells.py]: What circumstances could raise this catch-all exception handler? What testing could we do to raise to verify suitable responses?

File: craton/inventory/tests/unit/test_api.py [hunk 3]
```diff
@@ -40,6 +40,19 @@ def delete(self, path):
         return resp
 
 
+class APIV1CellsIDTest(APIV1Test):
+    @mock.patch.object(dbapi, 'cells_get_by_id')
+    def test_get_cells_by_id(self, mock_cells):
```
Comment (jimbaker) [inline craton/inventory/tests/unit/test_api.py]: Need testing for bad IDs

File: craton/inventory/tests/unit/test_api.py (hunk 1 above)
Comment (sulochan) [inline craton/inventory/tests/unit/test_api.py]: deleting non existing cells is not handled .. currently its 200. we will need to make separate PR to make that 404.

File: craton/inventory/api/v1/cells.py (hunk 2 above)
Comment (sulochan) [inline craton/inventory/api/v1/cells.py]: These would be db error, connection error, operations error etc ... causing 500 error. We could just test any thing other than notfound causes 500 but i am not sure what we gain from that. I think we should add specific errors that should be handled separately as we find them, but anything else is 500 for a user at this point really.

File: craton/inventory/tests/unit/test_api.py (hunk 3 above)
Comment (sulochan) [inline craton/inventory/tests/unit/test_api.py]: +1. ill add that.

**Comment by jimbaker** [comment]
+1, we can address my minor points later
